prov/util/src/util_mem_monitor.c

```diff
@@ -33,19 +33,28 @@
  */
 
 #include <ofi_mr.h>
-
+#include <ofi_mem_hooks.h>
 
 static struct ofi_uffd uffd;
 struct ofi_mem_monitor *uffd_monitor = &uffd.monitor;
 
+static struct ofi_patcher patcher;
+struct ofi_mem_monitor *patcher_monitor = &patcher.monitor;
 
 /*
  * Initialize all available memory monitors
  */
 void ofi_monitor_init(void)
 {
-	fastlock_init(&uffd_monitor->lock);
-	dlist_init(&uffd_monitor->list);
+	if (cache_params.core_monitor == 0) {
+		fastlock_init(&uffd_monitor->lock);
+		dlist_init(&uffd_monitor->list);
+	}
+
+	if (cache_params.core_monitor == 1) {
+		fastlock_init(&patcher_monitor->lock);
+		dlist_init(&patcher_monitor->list);
+	}
```

Comment:
We will want to initialize all monitors that we can here, regardless of which one is selected as the default.